### PR TITLE
[Finder] Add types to private properties

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | n0
| Deprecations? | n0
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Finder/Comparator/Comparator.php [hunk 1]
```diff
@@ -18,8 +18,8 @@
  */
 class Comparator
 {
-    private $target;
-    private $operator = '==';
+    private ?string $target = null;
```
Comment (stof) [inline src/Symfony/Component/Finder/Comparator/Comparator.php]: the type of the getter probably needs to be updated (or this should default to the empty string).

Comment (derrabus) [reply src/Symfony/Component/Finder/Comparator/Comparator.php]: Calling the getter before initialization does not really make sense. The two child classes of `Comparator` do initialize the property via the constructor. Let's skip the initialization.

Comment (derrabus) [reply src/Symfony/Component/Finder/Comparator/Comparator.php]: Let's make that class a bit more robust: #42137

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @derrabus.

**Pull request merged**